src/Sentry/Internal/Hub.cs

```diff
@@ -121,6 +125,34 @@ public ITransaction StartTransaction(string name, string operation)
                 transaction.Sdk.AddPackage(protocolPackageName, nameAndVersion.Version);
             }
 
+            // Make a sampling decision
+            var samplingContext = new TransactionSamplingContext(
+                context,
+                customSamplingContext
+            );
+
+            var sampleRate =
+                // Custom sampler may not exist or may return null, in which case we fallback
+                // to the static sample rate.
+                _options.TracesSampler?.Invoke(samplingContext)
```

Comment:
This is different from Java. I decided to allow the dynamic sampler to return `null` to fallback to the configured sample rate. I think it would be very useful, in case when you only want special sampling rules for a subset of transactions and default rules for everything else.

Reply:
It makes sense to me and I don't have a problem with that.
Maybe @marandaneto has opinions though.

Please make a note on the spec then that this is an alternative approach.

Reply:
that sounds a good use case, lets talk about this today in the meeting

Reply:
we'll make this on Java too